bokehjs/test/unit/models/renderers/graph_renderer.ts

```diff
@@ -9,34 +9,54 @@ import {Plot} from "@bokehjs/models/plots"
 import {Document} from "@bokehjs/document"
 import {build_view} from "@bokehjs/core/build_views"
 import {SerializationError} from "@bokehjs/core/serializer"
+import {XYGlyph} from "@bokehjs/models/glyphs/xy_glyph"
+import {Expression} from "@bokehjs/models"
+
+const layout_provider = new StaticLayoutProvider({
+  graph_layout: {
+    4: [2, 1],
+    5: [2, 2],
+    6: [3, 1],
+    7: [3, 2],
+  },
+})
+
+const node_renderer = new GlyphRenderer({
+  glyph: new Circle({size: 10, fill_color: "red"}),
+  data_source: new ColumnDataSource({data: {index: [4, 5, 6, 7]}}),
+})
+const edge_renderer = new GlyphRenderer({
+  glyph: new MultiLine({line_width: 2, line_color: "gray"}),
+  data_source: new ColumnDataSource({data: {start: [4, 4, 5, 6], end: [5, 6, 6, 7]}}),
+})
+
+const graph = new GraphRenderer({layout_provider, node_renderer, edge_renderer})
```

Comment:
This code needs to be wrapped in a function and fresh objects need to be produced for every test. This way we can make sure that tests can be run e.g. out-of-order, and there will be no mutable state creeping between tests.